ooni/oonicli.py

```diff
@@ -307,11 +345,163 @@ def post_director_start(_):
                                   collector)
         return director.allTestsDone
 
-    def start():
+
+    d.addCallback(setup_nettest)
+    d.addCallback(post_director_start)
+    d.addErrback(director_startup_handled_failures)
+    d.addErrback(director_startup_other_failures)
+    return d
```

Comment:
Note to self: have verified that new refactored `runWithDirector()` matches the old one